MapboxCoreNavigation/RouteController.swift

```diff
@@ -475,50 +475,78 @@ extension RouteController: CLLocationManagerDelegate {
             
             // Also only do one 'findFasterRoute' call per time
             self.isFindingFasterRoute = false
-
-            guard let route = mostSimilarRoute else {
-                return
-            }
-
-            guard let firstLeg = route.legs.first, let firstStep = firstLeg.steps.first else {
+            
+                
+            guard let route = mostSimilarRoute, let routes = routes else {
                 return
             }
             
-            // Check if the new route is faster by comparing the ETA to the current ETA. Should be 10% faster or more
-            let routeIsFaster = firstStep.expectedTravelTime >= RouteControllerMediumAlertInterval && currentUpcomingManeuver == firstLeg.steps[1] && route.expectedTravelTime <= 0.9 * durationRemaining
-            
+            self.applyNewRerouteIfNeeded(mostSimilarRoute: route, allRoutes: routes, currentUpcomingManeuver: currentUpcomingManeuver, durationRemaining: durationRemaining)
+        }
+    }
+    
+    func applyNewRerouteIfNeeded(mostSimilarRoute: Route, allRoutes: [Route], currentUpcomingManeuver: RouteStep, durationRemaining: TimeInterval) {
```

Comment:
New abstracted function that should be reviewed critically

Reply:
Can this function have private access?